### PR TITLE
Add `maze-maker` analyzer rules

### PR DESCRIPTION
**Pull request by jiegillet**
[Sister PR over here](https://github.com/exercism/website-copy/pull/2281).

Pretty standard stuff.
I also updated the dependencies.

### CODE REVIEW

**Review by ceddlyburge — APPROVED** [review]
Looks good :)

**Comment by jiegillet** [comment]
@exercism/maintainers-admin can I get an approval please? 💛

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
